utils/spdx/src/main/kotlin/Extensions.kt

```diff
@@ -30,32 +28,6 @@ import org.ossreviewtoolkit.utils.spdx.SpdxExpression.Strictness
 
 private val logger = loggerOf(MethodHandles.lookup().lookupClass())
```

Comment:
The extension functions are in similar relation to both SpdxOperator and SpdxExpression.
Moving the extension functions into the `SpdxOperator` class to me seems a bit odd in this regard.

What do you think about instead extracting each class contained in `SpdxExpression.kt` into its own dedicated class, for a better overview. But keep the extension functions where they currently are?

Reply:
> Moving the extension functions into the `SpdxOperator` class to me seems a bit odd in this regard.

Maybe it will become a bit less odd after some further refactoring that this is a preparation for:

I intend to split the current `spdx-utils` module into three modules:

* `spdx-license` (for SPDX licenses / exceptions / mapping)
* `spdx-expression` (for SPDX expression parsing)
* `spdx-document` (for SPDX document handling)
 
The new `SpdxOperator.kt` would be part of the `spdx-expression` module, so it's clear that it relates to both SpdxOperator and SpdxExpression.

> But keep the extension functions where they currently are?

I'd also like to achieve to 1) reduce the number of functions in that file in order to get rid of the `@file:Suppress("TooManyFunctions")` annotation, and 2) go a bit into the direction of what we discussed earlier and avoid generic a `Extensions.kt` in favor of files that group functions by purpose.

Reply:
To me, the extension functions in the operator file to me still seem to be closer related to the SpdxExpression class. Is there any reason why these cannot be class member functions? Turning them into member functions would also solve the problem of too many functions in `Extensions.kt` .

Reply:
For example if

    infix fun SpdxLicense.and(other: SpdxExpression) = SpdxCompoundExpression(toExpression(), SpdxOperator.AND, other)

would be a member function of `SpdxLicense`, it would mean that the upcoming "low-level" `spdx-license` module would need to depend on the "high-level" `spdx-expression` module. But it should be the other way around. That's why IMO using extension function makes sense here.

Reply:
My bad I misread the extension functions to be for `SpdxExpression` but they are for `SpdxLicense`.